packages/flutter/lib/src/material/page_transitions_theme.dart

```diff
@@ -1027,3 +1039,189 @@ class _ZoomExitTransitionPainter extends SnapshotPainter {
     super.dispose();
   }
 }
+
+// Zooms and fades a new page in, zooming out the previous page. This transition
+// is designed to match the Android Q activity transition.
+//
+// This was the historical implementation of the cacheless zoom page transition
+// that was too slow to run on the Skia backend. This is being benchmarked on
+// the Impeller backend so that we can improve performance enough to restore
+// the default behavior.
+class _ZoomPageTransitionNoCache extends StatelessWidget {
```

Comment:
The old implementation uses an ImageFilter.matrix to scale the individual pages without cause text relayout/snapping. The fallback uses a transform layer because its designed to work well if we can't apply an ImageFilter to content, which is common with platform views on iOS/Web.